library/modules/Textures.cpp

```diff
@@ -118,42 +150,66 @@ std::vector<TexposHandle> slice_tileset(SDL_Surface* surface, int tile_px_w, int
     return handles;
 }
 
-TexposHandle Textures::loadTexture(SDL_Surface* surface) {
+TexposHandle Textures::loadTexture(SDL_Surface* surface, bool reserved) {
     if (!surface)
         return 0; // should be some error, i guess
+    if (loading_state) {
+        ERR(textures).printerr("unable to load texture during game loading\n");
```

Comment:
I thought we were going to use the reserved buffer for this? e.g. instead of printing an error and returning 0, set `reserved = true` and continue on

Reply:
It will be hidden behavior.
Also  you said we can't touch raws vector, when game does. So this is short interval when textures can't be loaded.
We can make kinda delayed queue for such case. But i think it is too much:) who need load dynamic texposes in this particular short interval?

Reply:
I said we couldn't *extend* the raws vector. the buffer range will be safe to modify at all times

Reply:
stl vector not thread safe, are you sure?

Reply:
there isn't any concurrent access here, is there? we're running in DF's simulation thread, and you have locks protecting the Textures API from concurrent access from DFHack threads